integration-tests/src/test/resources/testng.xml

```diff
@@ -20,7 +20,7 @@
 <!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
 
 
-<suite name="IntegrationTestSuite">
+<suite name="IntegrationTestSuite" data-provider-thread-count="2">
```

Comment:
nit: might be helpful to add what this does / how to use in README.md

Reply:
It seems Travis didn't pick up my last commit due to an outage yesterday. I added some in README.

For `thread-count` and `data-provider-thread-count`, they are currently fixed for `IntegrationTestSuite` as far as I can tell ([command line parameters will be overridden by suite parameters](https://testng.org/doc/documentation-main.html#running-testng)). I think it could be better to set them via command line parameters, so that we can have more control over them when we want to run some tests locally.

Reply:
My thought behind this comment was to mention @DataProvider(parallel = true) and how that can be uses/works with data-provider-thread-count. Regarding, the fixed value, we can also move it to command line.

Reply:
Oh hmm, should we document those by ourselves? TestNG has already a couple of documents in their official document, Javadoc, and `testng.dtd` and I thought they would be enough.